concent_api/core/subtask_helpers.py

```diff
@@ -185,3 +186,22 @@ def are_subtask_results_accepted_messages_signed_by_the_same_requestor(subtask_r
         ) for subtask_results_accepted in subtask_results_accepted_list
     )
     return are_all_signed_by_requestor
+
+
+def get_one_or_none_subtasks_from_database(
+    subtask_id: str,
+    optional_condition_dict: Dict[str, str] = None,
+) -> Optional[Subtask]:
```

Comment:
I did it similar way at the beginning, but it was the cause of lot, very lot of IDE warnings. IDE don't know what type it is (generic function returns `-> object`) and shows warnings (e.g. some_object.subtask_id - IDE sees it like this). 

I would leave it like it is now, it is good in mostly cases, and if we would do something different we can do it manually with `model.objects.filter(something)`.. 

What do you think??

Reply:
I really think this function needs to be changed because it hides too much and is too much of a special case.

About warnings - surely there's some way around them. Can you show them to me? Python is a dynamic language. Doing things like returning arbitrary objects or duck typing by calling a method on the object whose type you don't know is nothing special.

Reply:
as discussed, done